### ISSUE TITLE
Can this output retina images?

### ISSUE THREAD

**Issue by georgecoltart** [mapbox/leaflet-image#9]
I noticed when on a retina display the source map is @2x but it exports a @1x. 
I don't know enough about how canvas works but using this plugin would it be possible to output an @2x image? 

thanks for your work on this it's great.

**Comment by tmcw** [comment]
Yes, though right now I'm pausing all development since #8 is a blocker for this being useful.

**Comment by robbiet480** [comment]
@tmcw Could you point me in the right direction to get retina support going?

**Comment by tmcw** [comment]
The [canvas would be 2x](https://github.com/mapbox/leaflet-image/blob/gh-pages/index.js#L10-L11) as large, and you would [scale the Point and tileSize math by 2](https://github.com/mapbox/leaflet-image/blob/gh-pages/index.js#L52-L121).

**Comment by robbiet480** [comment]
@tmcw Got it, thanks!

**Comment by baxford** [comment]
Hi @tmcw,
first off thanks for the tips about this.
I'm trying to implement the above, but I have a few questions. I am quite new to tiles, so apologies if these are dumb questions.
My understanding is that for a retina image, we need to display the same map but with double the amount of pixels - which is the reason for making the canvas 2x as large, which I have done. 
I have tried a couple of ways of getting the higher resolution image, but with only limited success so far:
1 - change tileSize to 512 and adjust the tile calculations accordingly. If this is the correct approach, do I need to adjust the map bounds (from  map.getPixelBounds()) or the tileJson bounds (https://github.com/mapbox/leaflet-image/blob/gh-pages/index.js#L70-L72) ?

You also mentioned that the Point math needs to be scaled by 2 - is that in the following lines:
https://github.com/mapbox/leaflet-image/blob/gh-pages/index.js#L85-L96 ? 

2 - For each tile that is requested, is it also possible to request 4 tiles at next higher zoom level? This would create problems at the  highest zoom level though so I guess it may not be a good option.

Anway, if it's possible to get a few more pointers on which values need adjusting it would be greatly appreciated. I'd be happy to contribute some work back into the project once I have it working.

thanks for any assistance you can provide!

**Comment by hjrobinson** [comment]
I tried this:

At lines 17 and 18:
`canvas.width = dimensions.x*2;
canvas.height = dimensions.y;*2`

At lines 82 and 83:
`canvas.width = dimensions.x*2;
canvas.height = dimensions.y*2;`

At lines 175 and 176:
`
ctx.drawImage(d.img, Math.floor(d.pos.x*2), Math.floor(d.pos.y*2),
d.size*2, d.size*2);
`

Seems to work. Correct me if I'm wrong.

**Comment by divyekhanna** [comment]
@hjrobinson The base map prints correctly, but the overlay items don't scale with this method.
